build/cpp/kvstore/src/main.cc

```diff
@@ -0,0 +1,28 @@
+#include "kvstore_server.h"
+
+DEFINE_string(store, "", "Stores and loads kvstore form .txt files");
+
+kvstore_server::KeyValueStoreServiceImpl *service;
+
+// calls service destructor upon shutdown signal
+void SignalHandler(int signum) {
+  LOG(INFO) << "Handling signal: " << signum;
+  delete service;
+  exit(signum);
+}
+
+// function for termination signals
+void sig() {
+  signal(SIGINT, SignalHandler);
+  signal(SIGTERM, SignalHandler);
+}
```

Comment:
Signal thread that catches the termination signals to store data upon termination.